lib/plug/conn.ex

```diff
@@ -192,6 +192,14 @@ defmodule Plug.Conn do
     """
   end
 
+  defmodule InvalidHeaderKeyFormat do
+    defexception message: "given header key has invalid format"
```

Comment:
We should rather say: "headers should be given in lowercase", otherwise the error message won't be super clear.